Feature/infinite list

philippe-cleany opened the pull request.

philippe-cleany committed:
package: add react-infinite-scroller

Add react-infinite-scroller package version 1.0.15.
We do this in order to create a new list component that will use
infinite scroll.

philippe-cleany committed:
list: add InfiniteList component

We add new component: <InfiniteList />.
this compoent render a scrollable list that load data if there still get
data. this component use react-infinite-scroller component.
We do this in order to have scrollable list in the app.

philippe-cleany committed:
reducer: change behavior of list reducer

We allow reducer to know when we use infiniteList or List component,
and the add couple of different action for the reducer.
We to this in order to load more data in case of infiniteList is query
the reducer.

philippe-cleany committed:
action: add infiniteList prop

We add inifiniteList prop in order to no if we use infiniteList or List
for query the right query for the user.

philippe-cleany committed:
Exemple: add infiniteList example

We add InfiniteList exmeple in order to view directely our component in
actions.

sebastien-cleany commented on e2e/tests/permissions.js:
```diff
@@ -12,11 +12,11 @@ describe('Permissions', () => {
         'http://localhost:8083/#/users/create'
     )(driver);
-    const EditPage = editPageFactory('http://localhost:8083/#/users/1/edit')(driver);
+    const EditPage = editPageFactory('http://localhost:8083/#/users/1')(driver);
```
nope, since your change on show/edit default.

philippe-cleany replied:
it should not be commited

sebastien-cleany commented on src/reducer/admin/resource/list/ids.js:
```diff
@@ -36,8 +36,17 @@ export default resource => (
-            return addRecordIds(payload.data.map(({ id }) => id), []);
+        case CRUD_GET_LIST_SUCCESS: {
+            if (meta.infiniteList) {
+                if (requestPayload.pagination.page === 1) {
```
if (!meta.infiniteList || requestPayload.pagination.page === 1) {
  return addRecord();
}

sebastien-cleany commented on src/mui/list/InfiniteList.js:
```diff
@@ -0,0 +1,216 @@
+});
+
+function mapStateToProps(state, props) {
+    const resourceState = state.admin.resources[props.resource];
```
See my previous comment

philippe-cleany replied:
I trying this way but it doesn't work, I propose to add hasmorre props to the list and import it to infiniteList component

philippe-cleany committed:
hotfix: e2e/tests/permissions

We commit e2e/tests/permissions.js from master.
But the file was change in local branch. We fix it by checkout the file
from local.

philippe-cleany committed:
ids: change request

apply change request.

philippe-cleany committed:
run lint

apply lint on :
e2e/tests/permissions.js
src/reducer/admin/resource/list/ids.js

philippe-cleany committed:
List: add hasMore props

The list component of AOR don't have hasMore props (that props defined
if there are more date to load). Now we add this props to the
mapStateToProps function. We do this in order to use it for the
infiniteList component and in order to have less code to write.

philippe-cleany committed:
code review: infiniteList

We have duplication of function that already write in list component.
Now we deleted some function, and we import mapStateToProps from list
we the custom component hasMore that usefull for the infiniteList
component. We do this in order to have less code to write.
